public/app/features/plugins/extensions/getPluginExtensions.ts

```diff
@@ -37,16 +37,22 @@ import {
 export const getObservablePluginExtensions = (
   options: Omit<GetExtensionsOptions, 'addedComponentsRegistry' | 'addedLinksRegistry'>
 ): Observable<ReturnType<GetExtensions>> => {
+  const { extensionPointId } = options;
+  const { addedComponentsRegistry, addedLinksRegistry } = pluginExtensionRegistries;
+
   return combineLatest([
-    pluginExtensionRegistries.addedComponentsRegistry.asObservable(),
-    pluginExtensionRegistries.addedLinksRegistry.asObservable(),
+    addedComponentsRegistry.asObservableSlice((state) => state[extensionPointId]),
+    addedLinksRegistry.asObservableSlice((state) => state[extensionPointId]),
   ]).pipe(
-    filter(([components, links]) => Boolean(components) && Boolean(links)), // filter out uninitialized registries
```

Comment:
Is this filtering moved somewhere else now?

Reply:
I'm don't recall exactly why we introduced this but I know it was some edge case, correct me if I'm wrong @leventebalogh